### PR TITLE
[Material] [Tizen] Adds Xamarin.Forms.Material.Tizen for Visual

### LINKED ISSUE #5008: [Material] Linear Progress Indicator
# Material Linear Progress Indicator

https://material.io/develop/ios/components/progress-indicators/progress-views/

- **Visually Identical**
   The rendering should be such that the naked eye can’t tell the difference at a glance. _Not to be confused with “pixel perfect”_

ProgressBar doesn't expose very many apis so the expectation here is that the default would size and color itself the same on iOS and Android

The hopefully not too tricky part here will be that there is only a Material ProgressBar iOS. On the android one it's still just using the original control so material visual on Android will just apply default styles or just inherit from the existing renderer

#### Current state
image

# Scenarios

```XAML
<ProgressBar Visual="Material" />
```

# Difficulty : medium

### LINKED ISSUE #5018: [Material] [Android, iOS] Slider Spec
# Material Slider 

https://material.io/design/components/sliders.html

- **Visually Identical**
   The rendering should be such that the naked eye can’t tell the difference at a glance. _Not to be confused with “pixel perfect”_

- **Performance**
Material renderer should be modeled after the fast renderer

Slider doesn't expose very many apis so the expectation here is that the default would size and color itself the same on iOS and Android

The hopefully not too tricky part will be that there is only a Material slider on iOS. On the android one it's still using the original control so material visual on Android will just apply default styles

#### Current state
image

# Scenarios

```XAML
<Slider Visual="Material" />
```

# Difficulty : medium possibly difficult

### PR DESCRIPTION
**Pull request by rookiejava**
### Description of Change ###

Added the Xamarin.Forms.Material.Tizen project for supporting Visual on Tizen. This PR includes the following material renderers. 
- Material ProgressBar
- Material Slider
- Material Frame 

### Issues Resolved ### 
- fixes for Tizen #5008 
- fixes for Tizen #5018

### API Changes ###
 None

### Platforms Affected ### 
- Tizen

### Behavioral/Visual Changes ###
New material renderers for tizen.

### Before/After Screenshots ### 
material_tizen

### Testing Procedure ###
Use `VisualGallery` inside `ControlGallery`.

### PR Checklist ###

- [ ] Has automated tests 
- [X] Rebased on top of the target branch at time of PR
- [X] Changes adhere to coding standard

### CODE REVIEW

File: Xamarin.Forms.Material.Android/MaterialColors.cs [hunk 1]
```diff
@@ -8,16 +8,23 @@
 using ASeekBar = Android.Widget.AbsSeekBar;
 using PlatformColor = Android.Graphics.Color;
 using Xamarin.Forms.Platform.Android;
-#else
+#elif __IOS__
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Material.Android/MaterialColors.cs]: are we sure about this ? iOS does uses this project ?

Comment (rookiejava) [reply Xamarin.Forms.Material.Android/MaterialColors.cs]: well... I think so. I just followed the existing condition. 😉

**Comment by rookiejava** [comment]
If this PR can be merged sooner, it will help us proceed with the next step. 😉

**Comment by PureWeen** [comment]
@rookiejava  The OSX build part of the CI is throwing the following errors
```
"/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj" (default target) (44) ->
  MaterialFrameRenderer.cs(8,12): error CS0246: The type or namespace name 'ExportRendererAttribute' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialFrameRenderer.cs(8,12): error CS0246: The type or namespace name 'ExportRenderer' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialProgressBarRenderer.cs(7,12): error CS0246: The type or namespace name 'ExportRendererAttribute' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialProgressBarRenderer.cs(7,12): error CS0246: The type or namespace name 'ExportRenderer' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialSliderRenderer.cs(7,12): error CS0246: The type or namespace name 'ExportRendererAttribute' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialSliderRenderer.cs(7,12): error CS0246: The type or namespace name 'ExportRenderer' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialFrameRenderer.cs(6,39): error CS0234: The type or namespace name 'Tizen' does not exist in the namespace 'Xamarin.Forms.Platform' (are you missing an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialFrameRenderer.cs(2,30): error CS0234: The type or namespace name 'Tizen' does not exist in the namespace 'Xamarin.Forms.Platform' (are you missing an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialProgressBarRenderer.cs(5,39): error CS0234: The type or namespace name 'Tizen' does not exist in the namespace 'Xamarin.Forms.Platform' (are you missing an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialProgressBarRenderer.cs(2,30): error CS0234: The type or namespace name 'Tizen' does not exist in the namespace 'Xamarin.Forms.Platform' (are you missing an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialSliderRenderer.cs(5,39): error CS0234: The type or namespace name 'Tizen' does not exist in the namespace 'Xamarin.Forms.Platform' (are you missing an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialSliderRenderer.cs(2,30): error CS0234: The type or namespace name 'Tizen' does not exist in the namespace 'Xamarin.Forms.Platform' (are you missing an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  Native/MBox.cs(3,30): error CS0234: The type or namespace name 'Tizen' does not exist in the namespace 'Xamarin.Forms.Platform' (are you missing an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  Native/MCanvas.cs(3,30): error CS0234: The type or namespace name 'Tizen' does not exist in the namespace 'Xamarin.Forms.Platform' (are you missing an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  /Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Android/MaterialColors.cs(18,30): error CS0234: The type or namespace name 'Tizen' does not exist in the namespace 'Xamarin.Forms.Platform' (are you missing an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialProgressBarRenderer.cs(10,45): error CS0246: The type or namespace name 'ProgressBarRenderer' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialSliderRenderer.cs(10,40): error CS0246: The type or namespace name 'SliderRenderer' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialFrameRenderer.cs(11,39): error CS0246: The type or namespace name 'ViewRenderer<,>' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  Native/MCanvas.cs(9,31): error CS0246: The type or namespace name 'IContainable<>' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  Native/MBox.cs(17,29): error CS0246: The type or namespace name 'LayoutEventArgs' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialProgressBarRenderer.cs(12,44): error CS0246: The type or namespace name 'ElementChangedEventArgs<>' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialSliderRenderer.cs(12,44): error CS0246: The type or namespace name 'ElementChangedEventArgs<>' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialFrameRenderer.cs(19,44): error CS0246: The type or namespace name 'ElementChangedEventArgs<>' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialSliderRenderer.cs(12,27): error CS0115: 'MaterialSliderRenderer.OnElementChanged(ElementChangedEventArgs<Slider>)': no suitable method found to override [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialFrameRenderer.cs(19,27): error CS0115: 'MaterialFrameRenderer.OnElementChanged(ElementChangedEventArgs<Frame>)': no suitable method found to override [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  MaterialProgressBarRenderer.cs(12,27): error CS0115: 'MaterialProgressBarRenderer.OnElementChanged(ElementChangedEventArgs<ProgressBar>)': no suitable method found to override [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]
  Native/MCanvas.cs(11,12): error CS0246: The type or namespace name 'ObservableCollection<>' could not be found (are you missing a using directive or an assembly reference?) [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Material.Tizen/Xamarin.Forms.Material.Tizen.csproj]

    90 Warning(s)
    28 Error(s)
```

**Comment by rookiejava** [comment]
@PureWeen It seems related #4646.  So, just like this, I've updated solution file (.sln) to make don't build it on `iPhone` and `iPhoneSimulator`. Could you check it again?

**Comment by PureWeen** [comment]
@rookiejava 

One more small fix it looks like
```
"/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.sln" (default target) (1) ->
"/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Platform.MacOS/Xamarin.Forms.Platform.MacOS.csproj" (default target) (33) ->
(CoreCompile target) -> 
  /Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Platform.iOS/Extensions/LabelExtensions.cs(111,4): error CS0618: 'NSLayoutManager.CharacterRangeForGlyphRange(NSRange, out NSRange)' is obsolete: 'Use 'GetCharacterRange' instead.' [/Users/vsts/agent/2.149.2/work/1/s/Xamarin.Forms.Platform.MacOS/Xamarin.Forms.Platform.MacOS.csproj]

    93 Warning(s)
    1 Error(s)
```

**Comment by rookiejava** [comment]
@PureWeen 
If I understand correctly, this error is resolved in #5613 and is not related to this PR. Do I need to rebase with the latest code?

**Comment by PureWeen** [comment]
@rookiejava yea if you could rebase that would be awesome!

It'd be ideal just to be sure this all compiles with the latest master build.

**Comment by rookiejava** [comment]
OK. I've just rebased.

**Comment by PureWeen** [comment]
@rookiejava 

```
The nuget command failed with exit code(1) and error(System.IO.DirectoryNotFoundException: Could not find a part of the path 'D:\a\1\s\Xamarin.Forms.Material.Tizen\bin\Debug\tizen40'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileSystemEnumerableIterator`1.CommonInit()
   at System.IO.FileSystemEnumerableIterator`1..ctor(String path, String originalUserPath, String searchPattern, SearchOption searchOption, SearchResultHandler`1 resultHandler, Boolean checkHost)
   at System.IO.Directory.GetFiles(String path, String searchPattern, SearchOption searchOption)
   at NuGet.Common.PathResolver.PerformWildcardSearch(String basePath, String searchPath, Boolean includeEmptyDirectories, String& normalizedBasePath)
   at NuGet.Packaging.PackageBuilder.ResolveSearchPattern(String basePath, String searchPath, String targetPath, Boolean includeEmptyDirectories)
   at NuGet.Packaging.PackageBuilder.AddFiles(String basePath, String source, String destination, String exclude)
   at NuGet.Packaging.PackageBuilder.PopulateFiles(String basePath, IEnumerable`1 files)
   at NuGet.Packaging.PackageBuilder.ReadManifest(Stream stream, String basePath, Func`2 propertyProvider)
   at NuGet.Packaging.PackageBuilder..ctor(String path, String basePath, Func`2 propertyProvider, Boolean includeEmptyDirectories)
   at NuGet.Commands.PackCommandRunner.CreatePackageBuilderFromNuspec(String path)
   at NuGet.Commands.PackCommandRunner.BuildFromNuspec(String path)
   at NuGet.CommandLine.PackCommand.ExecuteCommand()
   at NuGet.CommandLine.Command.ExecuteCommandAsync()
   at NuGet.CommandLine.Command.Execute()
   at NuGet.CommandLine.Program.MainCore(String workingDirectory, String[] args))
An error ocurred while trying to pack the files.
```

**Comment by rookiejava** [comment]
@PureWeen 
Uhm, I'm not sure but I don't think this PR directly cause this error. The modification of `nuspec` file doesn't look particularly strange. Could you check the configuration of the CI  server or internal build script to resolve path related problem? Thank you for your support. 😃

**Comment by PureWeen** [comment]
Right!! I remember needing to do this with the other material.

At some point we'll have this moved all over to YAML and then we should be able to do this with the PRs
I updated the CI server but can you update the YAML file in this pr??


basically you have to add these lines
```
Xamarin.Forms.Material.Tizen/bin/$(BuildConfigurations)/**/*.dll
Xamarin.Forms.Material.Tizen/bin/$(BuildConfigurations)/**/*.pdb
Xamarin.Forms.Material.Tizen/bin/$(BuildConfigurations)/**/*.mdb
```

to this file
https://github.com/xamarin/Xamarin.Forms/blob/master/build/steps/build-windows.yml#L61

**Comment by PureWeen** [comment]
🤞

**Comment by rookiejava** [comment]
🎉

**Pull request merged**